In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
USED=pd.read_csv('USED.csv',index_col=0)

In [2]:
USED.head(10)

RANKING                   GPU  NEW_PRICE  PASSMARK_SCORE  USED_PRICE  \
0      139       GeForce GTX 970     237.99          9673.0       76.20   
1      155         Radeon R9 390        NaN          9044.0       78.00   
2      146    GeForce GTX 780 Ti     339.99          9364.0       87.00   
3      136  GeForce GTX 1060 3GB     269.99          9733.0       91.75   
4      110       GeForce GTX 980     247.35         11247.0      125.00   
5       76    GeForce GTX 980 Ti     465.66         13869.0      162.45   
6      104      GeForce GTX 1660     229.99         11713.0      140.00   
7       82      GeForce GTX 1070     342.55         13494.0      168.00   
8      141        Radeon R9 Fury     443.10          9564.0      120.00   
9       70   GeForce GTX 1070 Ti     389.00         14577.0      189.97   

   USED_DOLLARS_PER_POINT  NEW_DOLLARS_PER_POINT  
0                0.007878               0.024604  
1                0.008625                    NaN  
2                0.009291               0.036308  
3                0.009427               0.027740  
4                0.011114               0.021993  
5                0.011713               0.033576  
6                0.011953               0.019635  
7                0.012450               0.025385  
8                0.012547               0.046330  
9                0.013032               0.026686

In [21]:
temp=USED.sort_values('RANKING',ascending=True)
temp=temp.reset_index().drop('index',axis=1)
list_=[]
for i in range(0,45):
    if temp['NEW_PRICE'][i] >0:
        if temp['USED_PRICE'][i] >0:
            list_.append(i)
temp=temp.loc[list_,:]
(temp['NEW_PRICE']/temp['USED_PRICE']).mean()

1.5678225571969637

### On average, new prices are 1.57x higher than used prices.
This will be factored in, as my max GPU cost is 200 dollars.

In [22]:
A=USED[USED['NEW_PRICE']<200]

A.sort_values('PASSMARK_SCORE',ascending=False)

RANKING                     GPU  NEW_PRICE  PASSMARK_SCORE  USED_PRICE  \
16       131  GeForce GTX 1650 SUPER     196.99         10055.0      153.26   
151      149       Radeon RX 6500 XT     167.99          9259.0         NaN   
21       150       Radeon RX 5500 XT     184.99          9191.0      150.00   
139      129                 NVS 315      69.99             NaN         NaN   

     USED_DOLLARS_PER_POINT  NEW_DOLLARS_PER_POINT  
16                 0.015242               0.019591  
151                     NaN               0.018143  
21                 0.016320               0.020127  
139                     NaN                    NaN

In [26]:
A=USED[USED['USED_PRICE']<200]

A.sort_values('PASSMARK_SCORE',ascending=False)

RANKING                     GPU  NEW_PRICE  PASSMARK_SCORE  USED_PRICE  \
9        70     GeForce GTX 1070 Ti     389.00         14577.0      189.97   
5        76      GeForce GTX 980 Ti     465.66         13869.0      162.45   
7        82        GeForce GTX 1070     342.55         13494.0      168.00   
12       91  GeForce GTX 1660 SUPER     219.99         12762.0      170.00   
23      102     GeForce GTX 1660 Ti     329.99         11808.0      195.00   
6       104        GeForce GTX 1660     229.99         11713.0      140.00   
4       110         GeForce GTX 980     247.35         11247.0      125.00   
19      127        Radeon R9 Fury X        NaN         10142.0      160.00   
10      130        GeForce GTX 1060     279.99         10068.0      132.45   
16      131  GeForce GTX 1650 SUPER     196.99         10055.0      153.26   
3       136    GeForce GTX 1060 3GB     269.99          9733.0       91.75   
0       139         GeForce GTX 970     237.99          9673.0       76.20   
24      140           Radeon RX 590     549.99          9664.0      159.99   
8       141          Radeon R9 Fury     443.10          9564.0      120.00   
26      142          Radeon R9 390X        NaN          9476.0      160.00   
2       146      GeForce GTX 780 Ti     339.99          9364.0       87.00   
21      150       Radeon RX 5500 XT     184.99          9191.0      150.00   
1       155           Radeon R9 390        NaN          9044.0       78.00   
13      158           Radeon RX 580     364.01          8918.0      120.00   

    USED_DOLLARS_PER_POINT  NEW_DOLLARS_PER_POINT  
9                 0.013032               0.026686  
5                 0.011713               0.033576  
7                 0.012450               0.025385  
12                0.013321               0.017238  
23                0.016514               0.027946  
6                 0.011953               0.019635  
4                 0.011114               0.021993  
19                0.015776                    NaN  
10                0.013156               0.027810  
16                0.015242               0.019591  
3                 0.009427               0.027740  
0                 0.007878               0.024604  
24                0.016555               0.056911  
8                 0.012547               0.046330  
26                0.016885                    NaN  
2                 0.009291               0.036308  
21                0.016320               0.020127  
1                 0.008625                    NaN  
13                0.013456               0.040817

In [27]:
14577.0/10055

1.4497265042267529

### The best performing 'used' card is 1.45x better than the best performing 'new' card, when capping at $200.

### On average, 'new' cards cost 1.57x more while, at my price point, being 1.45x slower than 'used'.

### This does not factor in the benefits of a warranty or ability to RMA a card dead on arrival.

## I am specifically looking for a card that performs at least 1.5x better than my 970.

In [29]:
USED.loc[0,'PASSMARK_SCORE']*1.5

14509.5

### I need a card that performs at 14509 Passmark score or higher, while also being under 200 dollars.

### There are no cards that fit this description in New, and only the 1070 ti fits the parameters .

In [48]:
pd.DataFrame(USED.loc[9,:].T)

9
RANKING                                  70
GPU                     GeForce GTX 1070 Ti
NEW_PRICE                             389.0
PASSMARK_SCORE                      14577.0
USED_PRICE                           189.97
USED_DOLLARS_PER_POINT             0.013032
NEW_DOLLARS_PER_POINT              0.026686